okio/src/main/java/okio/Buffer.java

```diff
@@ -1538,16 +1541,46 @@ public ByteString sha256() {
   private ByteString digest(String algorithm) {
     try {
       MessageDigest messageDigest = MessageDigest.getInstance(algorithm);
-      messageDigest.update(head.data, head.pos, head.limit - head.pos);
-      for (Segment s = head.next; s != head; s = s.next) {
-        messageDigest.update(s.data, s.pos, s.limit - s.pos);
+      if (head != null) {
+        messageDigest.update(head.data, head.pos, head.limit - head.pos);
+        for (Segment s = head.next; s != head; s = s.next) {
+          messageDigest.update(s.data, s.pos, s.limit - s.pos);
+        }
       }
       return ByteString.of(messageDigest.digest());
     } catch (NoSuchAlgorithmException e) {
       throw new AssertionError();
     }
   }
 
+  /** Returns the 160-bit SHA-1 HMAC of this buffer. */
+  public ByteString hmacSha1(ByteString key) {
+    return hmac("HmacSHA1", key);
+  }
+
+  /** Returns the 256-bit SHA-256 HMAC of this buffer. */
+  public ByteString hmacSha256(ByteString key) {
+    return hmac("HmacSHA256", key);
+  }
+
+  private ByteString hmac(String algorithm, ByteString key) {
```

Comment:
Right. I wasn't sure whether explicit checks were omitted intentionally. It's not without precedent in the codebase. But I had the impression explicit checks are the norm.
Is there a code style document where preferences for things like this are spelled out?